sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -576,6 +578,25 @@ trait Column extends DataFrame {
   override def as(alias: Symbol): Column = exprToColumn(Alias(expr, alias.name)())
 
   /**
+   * (Scala-specific) Explodes the column to zero or more rows by the provided function.
+   * {{{
+   *   val df = Seq(Tuple1("a b c"), Tuple1("d e")).toDataFrame("words")
+   *   val col = df("words")
+   *   col.explode {words: String => words.split(" ")}
+   * }}}
+   */
+  def explode[A, B : TypeTag](f: A => TraversableOnce[B]): Column = {
+    val dataType = ScalaReflection.schemaFor[B].dataType
+    val attributes = AttributeReference(schema.fields(0).name, dataType)() :: Nil
+    def rowFunction(row: Row) = {
+      f(row(0).asInstanceOf[A]).map(o => Row(ScalaReflection.convertToCatalyst(o, dataType)))
+    }
+    val generator = UserDefinedGenerator(attributes, rowFunction, expr :: Nil)
+    val plan = Generate(generator, join = true, outer = false, None, logicalPlan)
+    Column(sqlContext, Project(attributes, plan), attributes(0))
```

Comment:
i think this will break a lot of things because attributes(0) is different from the expression in project